### PR TITLE
More `@SuppressionWarnings` cleanup

### PR DESCRIPTION
**Pull request by markro49**

### CODE REVIEW

**Review by mernst — CHANGES_REQUESTED** [review]
In AtomicDouble.java, I was able to reduce the number of warning suppressions by removing them on a number of clients, and adding one on the method that all the clients call.

I removed the @PolyValue annotations on LongAdder that were technically true, but I don't see how they would ever be useful to a client, and they were not statically verifiable.

I fixed a Signedness Checker bug in https://github.com/typetools/checker-framework/pull/5862.  (This accounts for much of my delay in reviewing this; sorry about that.  Thank you for noticing the bug.)

When an explicit cast appears, I think it's better to write the type annotation on that cast.  By contrast, a suppression on the return value requires the reader to reason about what the effective cast type is and why it does not type-check.

File: guava/src/com/google/common/primitives/Booleans.java [hunk 1]
```diff
@@ -199,8 +198,7 @@ public static boolean contains(boolean[] array, boolean target) {
    * @param array the array to search for the sequence {@code target}
    * @param target the array to search for as a sub-sequence of {@code array}
    */
-  @SuppressWarnings("substringindex:return") // https://github.com/kelloggm/checker-framework/issues/206, 207 and 208
-  public static @LTEqLengthOf("#1") @SubstringIndexFor(value = "#1", offset = "#2.length - 1") int indexOf(boolean[] array, boolean[] target) {
+  public static @LTEqLengthOf("#1") int indexOf(boolean[] array, boolean[] target) {
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/Booleans.java:201]: src/com/google/common/primitives/Booleans.java:203: warning: [[index, substringindex, allcheckers]:anno.on.irrelevant] Annotation [@SubstringIndexFor(value = "#1", offset = "#2.length - 1")] is not applicable to int

File: guava/src/com/google/common/primitives/Bytes.java [hunk 2]
```diff
@@ -127,8 +127,7 @@ public static boolean contains(@PolySigned byte[] array, @PolySigned byte target
    * @param array the array to search for the sequence {@code target}
    * @param target the array to search for as a sub-sequence of {@code array}
    */
-  @SuppressWarnings("substringindex:return") // https://github.com/kelloggm/checker-framework/issues/206, 207 and 208
-  public static @LTEqLengthOf("#1") @SubstringIndexFor(value = "#1", offset = "#2.length - 1") int indexOf(@PolySigned byte[] array, @PolySigned byte[] target) {
+  public static @LTEqLengthOf("#1") int indexOf(@PolySigned byte[] array, @PolySigned byte[] target) {
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/Bytes.java:130]: src/com/google/common/primitives/Bytes.java:132: warning: [[index, substringindex, allcheckers]:anno.on.irrelevant] Annotation [@SubstringIndexFor(value = "#1", offset = "#2.length - 1")] is not applicable to int

File: guava/src/com/google/common/primitives/UnsignedInts.java [hunk 3]
```diff
@@ -106,10 +106,10 @@ public static int compare(@Unsigned int a, @Unsigned int b) {
    */
   // "(long value)" was originally "(@IntRange(from = 0, to = (2L << 32) - 1) long value)"
   // but that caused too many checker warnings.
-  @SuppressWarnings("signedness:cast.unsafe") // checkArgument guarantees value fits in Unsigned int
+  @SuppressWarnings("signedness:return") // needs Unsigned cast, but then cast.unsafe warning
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/UnsignedInts.java:109]: I agree - will revert change.

File: guava/src/com/google/common/primitives/UnsignedInteger.java [hunk 4]
```diff
@@ -188,16 +188,16 @@ public UnsignedInteger mod(UnsignedInteger val) {
    * will be equal to {@code this - 2^32}.
    */
   @Override
-  @SuppressWarnings("value:cast.unsafe") // Unknown int to PolyValue int is ok
+  @SuppressWarnings("value:return") // needs PolyValue cast, but then cast.unsafe warning
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/UnsignedInteger.java:191]: I agree - reverted change.

File: guava/src/com/google/common/primitives/Bytes.java [hunk 5]
```diff
@@ -229,11 +228,13 @@ public static boolean contains(@PolySigned byte[] array, @PolySigned byte target
    * @throws NullPointerException if {@code collection} or any of its elements is null
    * @since 1.0 (parameter was {@code Collection<Byte>} before 12.0)
    */
+  @SuppressWarnings("signedness:return") // needs PolySigned cast, but then cast.unsafe warning
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/Bytes.java:231]: This suppression is unneeded and will be removed.

File: guava/src/com/google/common/primitives/Bytes.java [hunk 6]
```diff
@@ -229,11 +228,13 @@ public static boolean contains(@PolySigned byte[] array, @PolySigned byte target
...
   public static <T extends Number> @PolySigned byte[] toArray(Collection<@PolySigned T> collection) {
     if (collection instanceof ByteArrayAsList) {
       return ((ByteArrayAsList) collection).toByteArray();
     }
 
+    @SuppressWarnings("signedness:assignment") // signedness is ok
```
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/Bytes.java:237]: This suppression is unneeded and will be removed.

File: guava/src/com/google/common/primitives/Bytes.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (markro49) [reply guava/src/com/google/common/primitives/Bytes.java:231]: added casts and changed comments

**Review by mernst — APPROVED** [review]

**Pull request merged**